src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -275,6 +275,14 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
 
             return new AspireStore(Path.Combine(aspireDir, ".aspire"));
         });
+        // Add configuration values from UserSecrets even in PublishMode which is
+        // typically invoked via the CLI which doesn't set the correct environment
+        // value to load up UserSecrets. We do this because for local deploy, we want
+        // to be able to use UserSecrets as storage for deployment state.
+        if (AppHostAssembly is not null && ExecutionContext.IsPublishMode)
+        {
+            _innerBuilder.Configuration.AddUserSecrets(AppHostAssembly);
```

Comment:
How does this work on CI? Does UserSecrets work when the current user doesn't have a $HOME defined?

Reply:
> How does this work on CI? 

This isn't intended to work in CI. We don't have a way of detecting whether we are in CI from the AppHost so we have to rely on some of the implicit behaviors. By default, `AddUserSecrets` will load the file as optional ([ref](https://source.dot.net/#Microsoft.Extensions.Configuration.UserSecrets/UserSecretsConfigurationExtensions.cs,85)) so it won't throw if not found. FWIW, we do have to verify this in the actual CI test app once it f lows through.